x-pack/metricbeat/module/aws/awshealth/awshealth.go

```diff
@@ -0,0 +1,437 @@
+	event := mb.Event{
+		MetricSetFields: mapstr.M{
+			"event_arn":                  getValueOrDefault(hd.event.Arn, ""),
+			"end_time":                   getValueOrDefault(hd.event.EndTime, time.Time{}),
+			"event_scope_code":           getValueOrDefault((*string)(&hd.event.EventScopeCode), ""),
+			"event_type_category":        getValueOrDefault((*string)(&hd.event.EventTypeCategory), ""),
+			"event_type_code":            getValueOrDefault(hd.event.EventTypeCode, ""),
+			"last_updated_time":          getValueOrDefault(hd.event.LastUpdatedTime, time.Time{}),
+			"region":                     getValueOrDefault(hd.event.Region, ""),
+			"service":                    getValueOrDefault(hd.event.Service, ""),
+			"start_time":                 getValueOrDefault(hd.event.StartTime, time.Time{}),
+			"status_code":                getValueOrDefault((*string)(&hd.event.StatusCode), ""),
+			"affected_entities_pending":  hd.affectedEntityPending,
+			"affected_entities_resolved": hd.affectedEntityResolved,
+			"affected_entities_others":   hd.affectedEntityOthers,
+			"affected_entities":          createAffectedEntityDetails(hd.affectedEntities),
+			"event_description":          hd.eventDescription,
+		},
+		RootFields: mapstr.M{
+			"cloud.provider": "aws",
+		},
+		ID: generateEventID(eventID),
+	}
+	return event
+}
+
+type HealthDetails struct {
+	event                  types.Event
+	eventDescription       string
+	affectedEntities       []types.AffectedEntity
+	affectedEntityPending  int32
+	affectedEntityResolved int32
+	affectedEntityOthers   int32
+}
+
+type AffectedEntityDetails struct {
+	AwsAccountId    string    `json:"aws_account_id"`
+	EntityUrl       string    `json:"entity_url"`
+	EntityValue     string    `json:"entity_value"`
+	LastUpdatedTime time.Time `json:"last_updated_time"`
+	StatusCode      string    `json:"status_code"`
+	EntityArn       string    `json:"entity_arn"`
+}
+
+// getValueOrDefault returns the dereferenced value of a pointer v of any type T.
+// If the pointer is nil, it returns the specified defaultValue of type T.
+func getValueOrDefault[T any](v *T, defaultValue T) T {
+	if v != nil {
+		return *v
+	}
+	return defaultValue
+}
+
+func createAffectedEntityDetails(affectedEntities []types.AffectedEntity) []AffectedEntityDetails {
+	aed := make([]AffectedEntityDetails, len(affectedEntities))
+	for i := range affectedEntities {
+		aed[i] = AffectedEntityDetails{
+			AwsAccountId:    getValueOrDefault(affectedEntities[i].AwsAccountId, ""),
+			EntityUrl:       getValueOrDefault(affectedEntities[i].EntityUrl, ""),
+			EntityValue:     getValueOrDefault(affectedEntities[i].EntityValue, ""),
+			LastUpdatedTime: getValueOrDefault(affectedEntities[i].LastUpdatedTime, time.Time{}),
+			StatusCode:      getValueOrDefault((*string)(&affectedEntities[i].StatusCode), ""),
+			EntityArn:       getValueOrDefault(affectedEntities[i].EntityArn, ""),
+		}
+	}
```

Comment:
```suggestion
        for _, affectedEntity := range affectedEntities {
	        aed = append(aed, AffectedEntityDetails{
		        AwsAccountId:    getValueOrDefault(affectedEntity.AwsAccountId, ""),
		        EntityUrl:       getValueOrDefault(affectedEntity.EntityUrl, ""),
		        EntityValue:     getValueOrDefault(affectedEntity.EntityValue, ""),
		        LastUpdatedTime: getValueOrDefault(affectedEntity.LastUpdatedTime, time.Time{}),
		        StatusCode:      getValueOrDefault((*string)(&affectedEntity.StatusCode), ""),
		        EntityArn:       getValueOrDefault(affectedEntity.EntityArn, ""),
	        })
        }
```

Reply:
please ignore empty spaces added by in suggestion.

Reply:
@shmsr , previously you mentioned the risks in the method suggested by the @kush-elastic and I moved to the code as we have now. Need your inputs on the above suggestion.

Reply:
@agithomas, i think that applies here as well.

Reply:
Yeah, same point as before.